File: src/python/test_gemini.py
import urllib.request
import json
import sys

API_KEY = "AIzaSyDwDCfv1cVVoUipZgngMAeMePMrqYSXMrI"
URL = f"https://generativelanguage.googleapis.com/v1beta/models?key={API_KEY}"

try:
    req = urllib.request.Request(URL)
    with urllib.request.urlopen(req) as response:
        data = json.loads(response.read().decode())
        valid_models = []
        for model in data.get('models', []):
            if 'generateContent' in model.get('supportedGenerationMethods', []):
                valid_models.append(model['name'])
        print("\n".join(valid_models))
except Exception as e:
    print(f"Error: {e}")

File: TestGemini.java
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.net.HttpURLConnection;
import java.net.URL;

public class TestGemini {
    public static void main(String[] args) {
        try {
            String apiKey = "AIzaSyDwDCfv1cVVoUipZgngMAeMePMrqYSXMrI";
            URL url = new URL("https://generativelanguage.googleapis.com/v1beta/models?key=" + apiKey);
            HttpURLConnection conn = (HttpURLConnection) url.openConnection();
            conn.setRequestMethod("GET");
            
            BufferedReader in = new BufferedReader(new InputStreamReader(conn.getInputStream()));
            String inputLine;
            StringBuilder response = new StringBuilder();
            while ((inputLine = in.readLine()) != null) {
                response.append(inputLine);
            }
            in.close();
            
            String json = response.toString();
            String[] parts = json.split("\"name\": \"");
            for (int i = 1; i < parts.length; i++) {
                String name = parts[i].substring(0, parts[i].indexOf("\""));
                if (name.contains("gemini")) {
                    System.out.println(name);
                }
            }
        } catch (Exception e) {
            e.printStackTrace();
        }
    }
}
